app/mailers/user_mailer.rb

```diff
@@ -1,8 +1,15 @@
 class UserMailer < ApplicationMailer
-  def authentication
-    @authentication = params[:authentication]
-    @user = @authentication.user
+  before_action :set_user
 
-    mail to: @user.email_address, subject: "Sign in to Hackathons"
+  def authentication(authentication = nil)
+    @authentication = authentication || @user.authentications.last
```

Comment:
```suggestion
  def authentication(authentication)
    @authentication = authentication
```

Reply:
Yeah, we should favor explicity with this